File: 프로그래머스/3/43162. 네트워크/네트워크.py
def dfs(y, x, computers):
    global used
    used[y][x] = 1
    for dy in range(len(computers)):
        if computers[x][dy] == 1 and used[x][dy] == 0:
            used[x][dy] = 1
            dfs(x,dy,computers)

def solution(n, computers):
    global used
    answer = 0
    used = [[0] * n for _ in range(n)]
    for y in range(n):
        for x in range(n):
            if used[y][x] == 0 and computers[y][x] == 1:
                dfs(y, x, computers)
                answer += 1
    return answer

File: 프로그래머스/3/43162. 네트워크/네트워크.js
function solution(n, computers) {
    const used = Array(n).fill(false);
    const dfs = (start) => {
        const stack = [start];
        used[start] = true;
        while (stack.length) {
            const cur = stack.pop()
            for (let nxt = 0; nxt < n; nxt++) {
                if (cur === nxt) continue;
                if (computers[cur][nxt] === 1 && !used[nxt]) {
                    used[nxt] = true;
                    stack.push(nxt)
                }
            }
        }
    };
    
    let cnt = 0;
    for (let i = 0; i < n; i++) {
        if (!used[i]) {
            dfs(i);
            cnt++;
        }
    }
    return cnt;
}